packages/skill-lesson/utils/build-stripe-checkout-path.ts

```diff
@@ -0,0 +1,34 @@
+import {z} from 'zod'
+import {omitBy, isNil} from 'lodash'
+
+const ParamsSchema = z
+  .object({
+    productId: z.string().optional(),
+    couponId: z.string().optional(),
+    bulk: z.boolean(),
+    quantity: z.number().default(1),
+    userId: z.string().optional(),
+    upgradeFromPurchaseId: z.string().optional(),
+    cancelUrl: z.string().optional(),
+  })
+  .transform((params) => {
+    return {
+      ...params,
+      bulk: params.bulk.toString(),
+      productId: params.productId || '',
+      quantity: String(params.quantity),
+    }
+  })
+type Params = z.input<typeof ParamsSchema>
+
+export const buildStripeCheckoutPath = (params: Params) => {
+  const result = ParamsSchema.safeParse(params)
+  if (result.success) {
+    const queryParams = omitBy(result.data, isNil)
+    const queryParamString = new URLSearchParams(queryParams).toString()
+    return `/api/skill/checkout/stripe?${queryParamString}`
+  }
```

Comment:
This is a helper that we can use across apps to build the skill-api path for a Stripe checkout. It provides types and uses `URLSearchParams` to build the query param string. This means each app doesn't have to do a bunch of manual string interpolation and can instead just pass in a type-checked object.

Reply:
would love to see a test for this one :)